driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -368,16 +368,22 @@ public void onSuccess(Void result) {
                     @Override
                     public void onFailure(Throwable t) {
                         if (t instanceof UnsupportedProtocolVersionException) {
-                            cluster.manager.logUnsupportedVersionProtocol(host, ((UnsupportedProtocolVersionException) t).getUnsupportedVersion());
-                            cluster.manager.triggerOnDown(host, false);
+                            UnsupportedProtocolVersionException e = (UnsupportedProtocolVersionException) t;
+                            cluster.manager.logUnsupportedVersionProtocol(host, e.getUnsupportedVersion());
                         } else if (t instanceof ClusterNameMismatchException) {
                             ClusterNameMismatchException e = (ClusterNameMismatchException) t;
                             cluster.manager.logClusterNameMismatch(host, e.expectedClusterName, e.actualClusterName);
-                            cluster.manager.triggerOnDown(host, false);
                         } else {
-                            logger.warn("Error creating pool to " + host, t);
+                            logger.error("Error creating pool to " + host, t);
+                        }
+                        boolean fatal = t instanceof UnsupportedProtocolVersionException ||
+                                t instanceof ClusterNameMismatchException;
+                        cluster.manager.triggerOnDown(host, !fatal);
+                        if (t instanceof Error) {
```

Comment:
This is not strictly related to JAVA-1431, but we agreed that errors should be propagated.